### PR TITLE
Extract non SBT project

### PR DESCRIPTION
**Pull request by nightscape**
I am planning to add similar functionality for reading POMs to Mill.
It would be helpful to reuse as much code as possible from this project, so I split out the SBT-independent stuff into an extra project that I can use in Mill.
I hope this is ok in principle with you.
If so, I'd be happy for any feedback / improvement proposals :smiley:

### CODE REVIEW

**Comment by blast-hardcheese** [comment]
I have no objections to this; are you planning on depending on sbt-pom-reader from mill-pom-reader or similar?

**Comment by nightscape** [comment]
It would be great if the non-SBT part could be published independently.
I've set up the SBT build of sbt-pom-reader so that this is a separate project.
Right now I have `publishLocal`ed this project and am writing the corresponding Mill code.
So far it's working smoothly.

**Comment by blast-hardcheese** [comment]
Ah, I missed that in the refactor -- I see `pom-model` now. Do you anticipate any further changes to this project?

**Comment by nightscape** [comment]
I just went through the code again to check if more can be extracted to the `pom-model` project, but the stuff that's still in the `plugin` project is quite specific to SBT.
So I think there wouldn't be much more changes upcoming.
Are the project names and Maven coordinates ok like they are?
I put the `pom-model` under the organization `com.typesafe` because it's independent of SBT.
Otoh, only members of the `com.typesafe` organization can then publish the artifacts to Maven central afaik.

**Comment by nightscape** [comment]
@blast-hardcheese Ping 😉

**Comment by blast-hardcheese** [comment]
@nightscape Hey, sorry for dropping the ball here. I have no objections, I can merge but I'm not sure I have permissions to release though. I'll most likely be able to proceed later this week.

**Comment by nightscape** [comment]
Ok. If it turns out to be a problem in the `com.typesafe` namespace, we can also move the non-SBT stuff to `com.typesafe.sbt`. Might lead to minor confusions, but most people won't see this anyway.

**Comment by nightscape** [comment]
Hi @blast-hardcheese, do you have any news regarding publishing permissions?
If you don't have them, maybe @eed3si9n could chime in?

File: build.sbt [hunk 1]
```diff
@@ -17,21 +9,33 @@ val mvnEmbedderDeps = Seq(
...
-libraryDependencies ++= Seq(
-  "org.apache.maven" % "maven-embedder" % mvnVersion exclude("com.google.guava", "guava") exclude("org.codehaus.plexus", "plexus-utils"),
-  "org.apache.maven.resolver" % "maven-resolver-connector-basic" % mvnResolver,
-  "org.apache.maven.resolver" % "maven-resolver-transport-file" % mvnResolver,
-  "org.apache.maven.resolver" % "maven-resolver-transport-http" % mvnResolver,
-  "org.apache.maven.resolver" % "maven-resolver-transport-wagon" % mvnResolver,
-) ++ mvnEmbedderDeps
+lazy val pom = project.settings(
+  name := "pom-model",
+  organization := "com.typesafe",
```
Comment (eed3si9n) [inline build.sbt]: If we are changing organization name, let's not do com.typesafe.
Are you planning to publish this to Sonatype OSS?

Comment (eed3si9n) [reply build.sbt]: sbt plugins are published to Bintray under my control, so I can allow whoever to publish under whatever organization name. Sonatype OSS on the other hand requires authorization per organization name.

**Comment by gemelen** [comment]
It seems not relevant anymore (for a long time) to the originating project, so I want to close this particular PR as obsolete.

I like the idea itself of having a build tool independent, intermediary project presentation, but likely there is no demand for it.

@nightscape sorry to ping you for the 6-year old issue, bu may I have you confirmation? :)

**Comment by nightscape** [comment]
Yes, confirmed 👍
I'm not encountering many Maven projects ATM.

**Pull request closed without merging**